crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -141,29 +140,42 @@ impl ReadView {
     pub fn transaction(&self, tx_id: &TxId) -> StorageResult<Transaction> {
         let result = self.on_chain.transaction(tx_id);
         if result.is_not_found() {
-            if let Some(tx) = self.off_chain.old_transaction(tx_id)? {
-                Ok(tx)
-            } else {
-                Err(not_found!(Transactions))
-            }
+            self.off_chain.old_transaction(tx_id)
         } else {
             result
         }
     }
 
-    pub async fn transactions(
-        &self,
-        tx_ids: Vec<TxId>,
-    ) -> Vec<StorageResult<Transaction>> {
-        // TODO: Use multiget when it's implemented.
-        //  https://github.com/FuelLabs/fuel-core/issues/2344
-        let result = tx_ids
+    pub async fn transactions(&self, tx_ids: &[TxId]) -> Vec<StorageResult<Transaction>> {
+        let on_chain_results: BTreeMap<_, _> = tx_ids
```

Comment:
It is nice that this function handles this hard use case.

In most cases, we have only `on_chain` data unless we did regenesis. It would be nice if this did not affect performance for the primary use case.

Can we first request values from the on-chain database, and if any of them are not found, we will fall back into the logic with `enumerate` and `BTreeMap`?

Reply:
Good idea, I'll do it.

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2396/commits/5de80b77b587eed062f3dcfc5769f0d878f7dc3b